source/textUtils/uniscribe.py

```diff
@@ -0,0 +1,40 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2024 NV Access Limited, Leonard de Ruijter
+
+"""Wrapper functions for NVDAHelper uniscribe functions."""
+
+import ctypes
+from typing import Generator
+
+
+def splitAtCharacterBoundaries(text: str) -> Generator[str, None, None]:
+	"""
+	Splits a given string into real visible characters (or glyphs), thereby respecting character boundaries.
+	Contrary to just iterating over a string, this respects surrogate pairs, decomposite characters, etc.
+	"""
+	import NVDAHelper  # Import late to avoid circular import.
+
+	if not NVDAHelper.localLib:
+		raise RuntimeError("NVDAHelper not initialized")
+	if not text:
+		return
+	# uniscribe does some strange things

```

Comment:
is there an issue reference for this?

Reply:
I'm afraid not. I copied this from `OffsetsTextInfo._calculateUniscribeOffsets` where a similar workaround was applied for similar code.